packages/core/database/src/utils/identifiers/shortener.ts

```diff
@@ -29,14 +26,19 @@ export const HASH_SEPARATOR = ''; // no separator is needed, we will just attach
 export const IDENTIFIER_SEPARATOR = '_';
 export const MIN_TOKEN_LENGTH = 3;
 
-type NameToken = {
+export type NameToken = {
```

Comment:
Could we split this as
```ts
export type CompressibleNameToken = {
  compressible: true;
  name: string;
  allocatedLength?: number;
}

export type IncompressibleNameToken = { 
  compressible: false;
  name: string;
  allocatedLength?: number;
  shortName?: string;
}

export type NameToken = CompressibleNameToken | IncompressibleNameToken
```

So that we can discriminate on the compressible attribute and avoid making mistakes by using the shortName value for compressible tokens?

Reply:
Gladly, thanks!